src/anchoredinterval.jl

```diff
@@ -1,3 +1,7 @@
+abstract type Anchor end
+struct Ending <: Anchor end
+struct Beginning <: Anchor end
```

Comment:
I think I can re-use `Direction` here. I wrote this code before endpoints were introduced.